python/ray/train/torch.py

```diff
@@ -144,3 +139,135 @@ def on_shutdown(self, worker_group: WorkerGroup,
+    rank = train.local_rank()
+
+    if not torch.cuda.is_available():
+        device = torch.device("cpu")
+    else:
+        device = torch.device(f"cuda:{rank}")
+
+    return device
+
+
+def prepare_model(
+        model: torch.nn.Module,
+        move_to_device: bool = True,
+        wrap_ddp: bool = True,
+        ddp_kwargs: Optional[Dict[str, Any]] = None) -> torch.nn.Module:
+    """Prepares the model for distributed execution.
+
+    This allows you to use the same exact code regardless of number of
+    workers or the device type being used (CPU, GPU).
+
+    Args:
+        model (torch.nn.Module): A torch model to prepare.
+        move_to_device (bool): Whether to move the model to the correct
+            device. If set to False, the model needs to manually be moved
+            to the correct device.
+        wrap_ddp (bool): Whether to wrap models in
+            ``DistributedDataParallel``.
+        ddp_kwargs (Dict[str, Any]): Args to pass into
+            ``DistributedDataParallel`` initialization if ``wrap_ddp`` is
+            set to True.
+    """
+    ddp_kwargs = ddp_kwargs or {}
+
+    rank = train.local_rank()
+
+    device = get_device()
+
+    if torch.cuda.is_available():
+        torch.cuda.set_device(device)
+
+    if move_to_device:
+        logger.info(f"Moving model to device: {device}")
+        model = model.to(device)
+    if wrap_ddp and train.world_size() > 1:
+        logger.info("Wrapping provided model in DDP.")
+        if torch.cuda.is_available():
+            model = DistributedDataParallel(
+                model, device_ids=[rank], output_device=rank, **ddp_kwargs)
+        else:
+            model = DistributedDataParallel(model, **ddp_kwargs)
+
+    return model
+
+
+def prepare_data_loader(data_loader: torch.utils.data.DataLoader,
+                        add_dist_sampler: bool = True,
+                        move_to_device: bool = True) -> \
+        torch.utils.data.DataLoader:
+    """
+    Prepares DataLoader for distributed execution.
+
+    This allows you to use the same exact code regardless of number of
+    workers or the device type being used (CPU, GPU).
+
+    Args:
+        data_loader (torch.utils.data.DataLoader): The DataLoader to
+            prepare.
+        add_dist_sampler (bool): Whether to add a DistributedSampler to
+            the provided DataLoader.
+        move_to_device (bool): If set, automatically move the data
+            returned by the data loader to the correct device.
+    """
+
+    if train.world_size() > 1 \
+        and not isinstance(data_loader.sampler, DistributedSampler) \
+        and not (hasattr(data_loader, "dataset")
+                 and isinstance(data_loader.dataset, IterableDataset)) \
+            and add_dist_sampler:
+
+        def with_sampler(loader):
+            # Automatically set the DistributedSampler
+
+            shuffle = not isinstance(loader.sampler, SequentialSampler)
```

Comment:
Add a comment explaining this? Also the reason for why we set `shuffle` to False in `data_loader_args` in relation to `DistributedSampler`.

Reply:
Done!